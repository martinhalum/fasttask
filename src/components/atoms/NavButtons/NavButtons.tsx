/**
 *
 * NavButtons
 *
 */

import React from 'react';
import Icon from 'react-native-vector-icons/FontAwesome5';
import {TouchableOpacity, View} from 'react-native';

import NavButtonsStyles from './styles';

import type {PropsType} from './types';

function NavButtons({
  onLeftPress,
  onRightPress,
}: PropsType): React.ReactElement {
  return (
    <View style={NavButtonsStyles.wrapper}>
      <TouchableOpacity onPress={onLeftPress}>
        <Icon name="chevron-left" size={18} />
      </TouchableOpacity>
      <TouchableOpacity onPress={onRightPress}>
        <Icon name="edit" size={18} />
      </TouchableOpacity>
    </View>
  );
}

export default NavButtons;
