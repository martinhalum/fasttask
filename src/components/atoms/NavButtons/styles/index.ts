/**
 *
 * NavButtons Styles
 *
 */

import {StyleSheet} from 'react-native';

const NavButtonsStyles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});

export default NavButtonsStyles;
