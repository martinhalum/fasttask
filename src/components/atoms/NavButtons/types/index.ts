/**
 *
 * NavButtons Types
 *
 */

export type PropsType = {
  onLeftPress?: () => void;
  onRightPress?: () => void;
};
