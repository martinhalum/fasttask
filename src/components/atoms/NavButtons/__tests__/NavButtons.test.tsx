/**
 *
 * NavButtons Test
 *
 */

import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import NavButtons from '../NavButtons';

describe('NavButtons', () => {
  // Renders two buttons wrapped in a View
  it('should render two buttons wrapped in a View', () => {
    // Arrange
    const onLeftPress = jest.fn();
    const onRightPress = jest.fn();

    // Act
    const screen = render(
      <NavButtons onLeftPress={onLeftPress} onRightPress={onRightPress} />,
    );

    // Assert
    expect(screen.getByTestId('leftButton')).toBeTruthy();
    expect(screen.getByTestId('rightButton')).toBeTruthy();
    expect(screen.getByTestId('wrapper')).toBeTruthy();
  });

  // Optional onLeftPress and onRightPress callbacks can be provided
  it('should execute onLeftPress callback when left button is pressed', () => {
    // Arrange
    const onLeftPress = jest.fn();
    const onRightPress = jest.fn();

    // Act
    const screen = render(
      <NavButtons onLeftPress={onLeftPress} onRightPress={onRightPress} />,
    );
    fireEvent.press(screen.getByTestId('leftButton'));

    // Assert
    expect(onLeftPress).toHaveBeenCalled();
  });

  // onLeftPress and onRightPress are not provided
  it('should not execute onLeftPress or onRightPress callbacks when buttons are pressed', () => {
    // Arrange
    const onLeftPress = jest.fn();
    const onRightPress = jest.fn();

    // Act
    const screen = render(<NavButtons />);
    fireEvent.press(screen.getByTestId('leftButton'));
    fireEvent.press(screen.getByTestId('rightButton'));

    // Assert
    expect(onLeftPress).not.toHaveBeenCalled();
    expect(onRightPress).not.toHaveBeenCalled();
  });

  // onLeftPress and onRightPress are null
  it('should not execute onLeftPress or onRightPress callbacks when buttons are pressed', () => {
    // Arrange
    const onLeftPress = jest.fn();
    const onRightPress = jest.fn();

    // Act
    const screen = render(
      <NavButtons onLeftPress={null} onRightPress={null} />,
    );
    fireEvent.press(screen.getByTestId('leftButton'));
    fireEvent.press(screen.getByTestId('rightButton'));

    // Assert
    expect(onLeftPress).not.toHaveBeenCalled();
    expect(onRightPress).not.toHaveBeenCalled();
  });

  // onLeftPress and onRightPress are not functions
  it('should not execute onLeftPress or onRightPress callbacks when buttons are pressed', () => {
    // Arrange
    const onLeftPress = jest.fn();
    const onRightPress = jest.fn();

    // Act
    const screen = render(
      <NavButtons onLeftPress={123} onRightPress={'abc'} />,
    );
    fireEvent.press(screen.getByTestId('leftButton'));
    fireEvent.press(screen.getByTestId('rightButton'));

    // Assert
    expect(onLeftPress).not.toHaveBeenCalled();
    expect(onRightPress).not.toHaveBeenCalled();
  });
});
