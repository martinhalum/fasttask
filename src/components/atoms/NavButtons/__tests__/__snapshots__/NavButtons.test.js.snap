// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButtons should render NavButtons component with no buttons when onLeftPress and onRightPress are not provided 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 18,
          },
          undefined,
          {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 18,
          },
          undefined,
          {
            "fontFamily": "FontAwesome5Free-Regular",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {
            "fontWeight": "400",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`NavButtons should render NavButtons component with only left button when onRightPress is not provided 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 18,
          },
          undefined,
          {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 18,
          },
          undefined,
          {
            "fontFamily": "FontAwesome5Free-Regular",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {
            "fontWeight": "400",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
